src/Microsoft.Health.Fhir.Shared.Api/Features/Resources/Bundle/BundleHandler.cs

```diff
@@ -167,6 +242,18 @@ private async Task ExecuteRequests(Hl7.Fhir.Model.Bundle responseBundle, Hl7.Fhi
                         if (entryComponentResource.ResourceType == ResourceType.OperationOutcome)
                         {
                             entryComponent.Response.Outcome = entryComponentResource;
+
+                            if (responseBundle.Type == Hl7.Fhir.Model.Bundle.BundleType.TransactionResponse)
+                            {
+                                _bundleProcessingStatus = BundleProcessingStatus.Failed;
```

Comment:
Could we just throw the transaction exception here and let it bubble up?

Reply:
Actually we can. Not sure why that never occurred to me. And that emoji is my reaction after I found out it can be done this way :)